### PR TITLE
added data_tailscale_acl data source

### LINKED ISSUE #180: data resource for ACL
**Is your feature request related to a problem? Please describe.**
We would like to manage our ACLs across multiple terraform workspaces.
So we would like to read the ACL each time before we add a new customer and merge new rules to each.

**Describe the solution you'd like**
```
data "tailscale_acl" "self" {}
```

### PR DESCRIPTION
**Pull request by markwellis**
**What this PR does / why we need it**:
there is currently no acl data_source which makes working with the acl hard

**Which issue this PR fixes** *(use `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 

Fixes #180

**Special notes for your reviewer**:
I decided to expose it as a json blob as writing a schema and mangling it into that schema at load time was a lot of work and I don't see the benefit. 
`resource "tailscale_acl"` takes a json string so it doesn't seem unreasonable that the data source for that would give a json string too
Also the external script I've been using up till now also works by providing a json string, which makes this a drop in replacement for that.

It works quite well with this test

```hcl
data "tailscale_acl" "acl" {}

locals {
  acl = jsondecode(data.tailscale_acl.acl.json)
}

resource "tailscale_acl" "acl" {
  acl = jsonencode(merge(local.acl, {groups = merge(local.acl.groups, {"group:foo" = ["br@asd.com"]})}))

  allow_overwrite = true # in #303 
}
```

P.s. sorry if the code's not great, it's the first go I've written so I have no idea about best practice etc

### CODE REVIEW

File: tailscale/data_source_acl.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/tailscale/tailscale-client-go/tailscale"
+)
+
+func dataSourceAcl() *schema.Resource {
+	return &schema.Resource{
+		Description: "The acl data source gets the acl in a tailnet",
```
Comment (knyar) [inline tailscale/data_source_acl.go:15]:
```suggestion
		Description: "The acl data source gets the Tailscale ACL for a tailnet",
```

File: tailscale/data_source_acl.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/tailscale/tailscale-client-go/tailscale"
+)
+
+func dataSourceAcl() *schema.Resource {
```
Comment (knyar) [inline tailscale/data_source_acl.go:13]:
```suggestion
func dataSourceACL() *schema.Resource {
```

to match Go style

File: tailscale/data_source_acl.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+			"json": {
+				Computed:    true,
+				Type:        schema.TypeString,
+				Description: "the complete acl json",
+			},
+		},
+	}
+}
+
+func dataSourceAclRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```
Comment (knyar) [inline tailscale/data_source_acl.go:27]:
```suggestion
func dataSourceACLRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

File: tailscale/data_source_acl.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+func dataSourceAcl() *schema.Resource {
+	return &schema.Resource{
+		Description: "The acl data source gets the acl in a tailnet",
+		ReadContext: dataSourceAclRead,
+		Schema: map[string]*schema.Schema{
+			"json": {
+				Computed:    true,
+				Type:        schema.TypeString,
+				Description: "the complete acl json",
```
Comment (knyar) [inline tailscale/data_source_acl.go:21]:
```suggestion
				Description: "The contents of Tailscale ACL as JSON",
```

**Review by knyar — COMMENTED** [review]
Thank you! Just a couple of minor comments here.

File: tailscale/data_source_acl.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/tailscale/tailscale-client-go/tailscale"
+)
+
+func dataSourceACL() *schema.Resource {
+	return &schema.Resource{
+		Description: "The acl data source gets the Tailscale ACL for a tailnet",
+		ReadContext: dataSourceAclRead,
```
Comment (knyar) [inline tailscale/data_source_acl.go:16]:
```suggestion
		ReadContext: dataSourceACLRead,
```

File: tailscale/provider.go [hunk 6]
```diff
@@ -80,6 +80,7 @@ func Provider(options ...ProviderOption) *schema.Provider {
 			"tailscale_device":  dataSourceDevice(),
 			"tailscale_devices": dataSourceDevices(),
 			"tailscale_4via6":   dataSource4Via6(),
+			"tailscale_acl":     dataSourceAcl(),
```
Comment (knyar) [inline tailscale/provider.go:83]:
```suggestion
			"tailscale_acl":     dataSourceACL(),
```

**Comment by knyar** [comment]
> anything you need from me to get this moving?

It seems that CI checks are still failing. In this case I think what's missing is running `go generate` (or `tfplugindocs` directly) to regenerate markdown docs. Thank you!

**Comment by markwellis** [comment]
I have updated the PR with `tfplugindocs`

**Pull request merged**